### PR TITLE
Tuist stats: command events [part1]

### PR DESCRIPTION
**Pull request by andreacipriani**
### Context

- Stats will be reported at https://stats.tuist.io/
- On this first PR, we wrap the existing `ParsableCommand` to collect the data and create a `CommandEvent`. The event is sent to a new framework `TuistAnalytics` that will be responsible to tag the event

### Future work
-  Send the payload the server
- Add documentation
- Add opt-out

### Screenshot

Screen Shot 2020-12-20 at 3 29 33 PM

### CODE REVIEW

File: Sources/TuistKit/Commands/TuistCommand.swift [hunk 1]
```diff
@@ -56,7 +56,10 @@ public struct TuistCommand: ParsableCommand {
             _exit(exitCode(for: error).rawValue)
         }
         do {
-            try command.run()
+            var trackableCommand = TrackableCommand(command: command)
+            try trackableCommand.run(completion: { _ in
```
Comment (andreacipriani) [inline Sources/TuistKit/Commands/TuistCommand.swift:60]: This architecture allows me to not change any specific command

File: Sources/TuistAnalytics/CommandEvent.swift [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    let name: String
+    let subcommand: String?
+    let duration: TimeInterval
+
+    // SessionEnv
+    let clientId: String
+    let tuistVersion: String
+    let swiftVersion: String
+    let macOSVersion: String
+}
```
Comment (RomainBoulay) [inline Sources/TuistAnalytics/CommandEvent.swift:13]: With Apple brining a new processor architecture, I think we should also track that?
```suggestion
    let processorArchitecture: String  
}
```

or? 

```suggestion
    let processorArchitecture: ProcessorArchitecture
}

enum ProcessorArchitecture {
case x86, appleSilicon
}
```

File: Sources/TuistAnalytics/CommandEvent.swift [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import Foundation
+
+public struct CommandEvent: Codable {
```
Comment (RomainBoulay) [inline Sources/TuistAnalytics/CommandEvent.swift:3]: I know you've discarded the idea for now, but I still feel that having additional details like: `number of target used in a tuist focus command` would be very interesting. Not a deal breaker for me but I thought I'd share that!

Comment (RomainBoulay) [reply Sources/TuistAnalytics/CommandEvent.swift:3]: In this spirit, I'd add something like this in this struct:

```swift
let properties: [String: String]
```

File: Sources/TuistKit/Commands/TuistCommand.swift (hunk 1 above)
Comment (RomainBoulay) [reply Sources/TuistKit/Commands/TuistCommand.swift:60]: Yep that's nice 👍 Elegant solution 👏

**Review by RomainBoulay — APPROVED** [review]
I added a couple of comments but LGTM 🍷

File: Sources/TuistAnalytics/CommandEvent.swift (hunk 2 above)
Comment (andreacipriani) [reply Sources/TuistAnalytics/CommandEvent.swift:38]: Done

File: Sources/TuistAnalytics/CommandEvent.swift (hunk 3 above)
Comment (andreacipriani) [reply Sources/TuistAnalytics/CommandEvent.swift]: Done

**Review by RomainBoulay — APPROVED** [review]
Solid work @andreacipriani 👏

**Pull request merged**